src/main/java/bhs/devilbotz/utils/ShuffleboardManager.java

```diff
@@ -46,24 +57,53 @@ public ShuffleboardManager() {
             .withWidget(BuiltInWidgets.kBooleanBox)
             .getEntry();
 
-    driveTab
-        .add("Auto Mode", autoModeChooser)
-        .withWidget(BuiltInWidgets.kComboBoxChooser)
-        .withPosition(0, 0)
-        .withSize(2, 1);
-
     initAutoModeChooser();
+    initAutoModePreferences();
   }
 
   private void initAutoModeChooser() {
-    autoModeChooser.setDefaultOption("Backup Short", AutonomousModes.BACKUP_SHORT);
+    autoModeChooser.addOption("Routine 1: Sit Still", AutonomousModes.SIT_STILL);
+    autoModeChooser.setDefaultOption("Routine 2: Mobility", AutonomousModes.MOBILITY);
+    autoModeChooser.addOption("Routine 3: Score and Mobility", AutonomousModes.SCORE_AND_MOBILITY);
+    autoModeChooser.addOption("Routine 4: Dock and Engage", AutonomousModes.DOCK_AND_ENGAGE);
+    autoModeChooser.addOption(
+        "Routine 5: Mobility, Dock, and Engage", AutonomousModes.MOBILITY_DOCK_AND_ENGAGE);
+    autoModeChooser.addOption(
+        "Routine 6: Score, Dock, and Engage", AutonomousModes.SCORE_DOCK_AND_ENGAGE);
+    autoModeChooser.addOption(
+        "Routine 7: Score, Mobility, Dock, and Engage", AutonomousModes.SCORE_MOBILITY_DOCK_ENGAGE);
+    autoModeChooser.addOption(
+        "Routine 8: Score, Mobility, Pick, Dock, and Engage",
+        AutonomousModes.SCORE_MOBILITY_PICK_DOCK_ENGAGE);
+
+    autoModeChooser.addOption("Backup Short", AutonomousModes.BACKUP_SHORT);
     autoModeChooser.addOption("Backup and Balance", AutonomousModes.BACKUP_AND_BALANCE);
     autoModeChooser.addOption("Backup Far", AutonomousModes.BACKUP_FAR);
     autoModeChooser.addOption("Balance", AutonomousModes.BALANCE);
     autoModeChooser.addOption("Drive Distance", AutonomousModes.DRIVE_DISTANCE);
     autoModeChooser.addOption("Drive Distance PID", AutonomousModes.DRIVE_DISTANCE_PID);
-    autoModeChooser.setDefaultOption(
-        "Drive Distance Straight PID", AutonomousModes.DRIVE_STRAIGHT_DISTANCE_PID);
+    autoMode.add("Routine", autoModeChooser).withWidget(BuiltInWidgets.kComboBoxChooser);
+  }
+
+  private void initAutoModePreferences() {
+    autoDelay =
+        autoMode.add("Delay (in seconds)", 0).withWidget(BuiltInWidgets.kTextView).getEntry();
+
+    autoDistance =
+        autoMode.add("Distance (in meters)", 5).withWidget(BuiltInWidgets.kTextView).getEntry();
```

Comment:
Do these work or are they just theoretical for the future?

Reply:
autoDistance is currently used with the "Mobility" command.  E.g. you can specify the distance, select "Mobility" in the autonomous chooser and then enable autonomous mode.  Using @joshuman's `DriveStraughtPID` class, it should travel the distance requested in Shuffleboard. You can play with it in simulation.